File: Muslima/app(Muslima)/src/main/java/com/ebook/com/Models/MainActivity.java
package com.ebook.com.Models;

import android.content.Intent;
import android.media.Image;
import android.net.Uri;
import android.os.Bundle;
import android.view.Gravity;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.GravityCompat;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.ebook.com.Adapters.CategoryAdapter;
import com.ebook.com.R;
import com.ebook.com.databinding.ActivityMainBinding;
import com.google.android.material.navigation.NavigationView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding binding;
    NavigationView navigationView;
    DrawerLayout drawerLayout;
    View header;
    ArrayList<CategoryMode> list;
    CategoryAdapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        navigationView = (NavigationView) findViewById(R.id.navigationView);
        drawerLayout = (DrawerLayout)  findViewById(R.id.drawerLayout);
        binding.menu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (drawerLayout.isDrawerOpen(GravityCompat.START)){
                    drawerLayout.closeDrawer(GravityCompat.START);
                }
                else {
                    drawerLayout.openDrawer(GravityCompat.START);
                }
            }
        });
        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                    if (item.getItemId()==R.id.share){
                        String shareBody = "Hey! I am using the best eBook App" + "http://play.google.com/store/apps/details?id="+MainActivity.this.getPackageName();
                        Intent intent = new Intent(Intent.ACTION_SEND);
                        intent.setType("text/plain");
                        intent.putExtra(Intent.EXTRA_TEXT,shareBody);
                        startActivity(intent);
                        drawerLayout.closeDrawer(GravityCompat.START);
                    }
                    else if (item.getItemId()==R.id.rate) {
                            startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("http://play.google.com/store/apps/details?id="+MainActivity.this.getPackageName())));
                            drawerLayout.closeDrawer(GravityCompat.START);
                    }
                return false;
            }
        });


        navigationView.setNavigationItemSelectedListener(new NavigationView.OnNavigationItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                if (item.getItemId()==R.id.share){
                    String shareBody = "Hey! I am using the best eBook App" + "http://play.google.com/store/apps/details?id="+MainActivity.this.getPackageName();
                    Intent intent = new Intent(Intent.ACTION_SEND);
                    intent.setType("text/plain");
                    intent.putExtra(Intent.EXTRA_TEXT,shareBody);
                    startActivity(intent);
                    drawerLayout.closeDrawer(GravityCompat.START);
                }
                else if (item.getItemId()==R.id.rate) {
                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("http://play.google.com/store/apps/details?id="+MainActivity.this.getPackageName())));
                    drawerLayout.closeDrawer(GravityCompat.START);
                }
                return false;
            }
        });
        list = new ArrayList<>();
        list.add(new CategoryMode("chapter:1"));
        list.add(new CategoryMode("chapter:2"));
        list.add(new CategoryMode("chapter:3"));
        list.add(new CategoryMode("chapter:4"));
        list.add(new CategoryMode("chapter:5"));
        list.add(new CategoryMode("chapter:6"));
        list.add(new CategoryMode("chapter:7"));
        list.add(new CategoryMode("chapter:8"));
        list.add(new CategoryMode("chapter:9"));
        list.add(new CategoryMode("chapter:10"));

        LinearLayoutManager layoutManager = new LinearLayoutManager(this);
        binding.rvCategory.setLayoutManager(layoutManager);

        adapter = new CategoryAdapter(this, list);
        binding.rvCategory.setAdapter(adapter);
    }
}

File: Muslima/app(Muslima)/src/main/java/com/ebook/com/MainActivity.kt

//    class MainActivity : AppCompatActivity() {
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        enableEdgeToEdge()
//        setContentView(R.layout.activity_main)
//        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
//            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
//            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
//            insets
//        }
//    }
//}
//




//
//package com.ebook.com
//
//import android.os.Bundle
//import android.view.MenuItem
//import android.view.View
//import android.widget.ImageView
//import androidx.appcompat.app.AppCompatActivity
//import androidx.drawerlayout.widget.DrawerLayout
//import com.google.android.material.navigation.NavigationView
//import com.ebook.com.databinding.ActivityMainBinding  // Ensure this import exists
//import com.ebook.com.Models.CategoryMode  // Ensure correct package name
//
//class MainActivity : AppCompatActivity() {
//    private var binding: ActivityMainBinding? = null
//    private var navigationView: NavigationView? = null
//    private var drawerLayout: DrawerLayout? = null
//    private var menu: ImageView? = null
//    private var header: View? = null
//    private var list: ArrayList<CategoryMode>? = null
//
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//
//        // Inflate the layout using ViewBinding
//        binding = ActivityMainBinding.inflate(layoutInflater)
//        setContentView(binding?.root)
//
//        // Initialize UI components safely
//        drawerLayout = binding?.drawerLayout
//        navigationView = binding?.navigationView
//        menu = binding?.imageView2
//
//        // Get header from NavigationView
//        navigationView?.let {
//            header = it.getHeaderView(0)
//        }
//
//        // Initialize the list
//        list = ArrayList()
//
//        // Set up navigation menu
//        setupNavigationMenu()
//    }
//
//    private fun setupNavigationMenu() {
//        menu?.setOnClickListener {
//            if (drawerLayout?.isDrawerOpen(navigationView!!) == true) {
//                drawerLayout?.closeDrawer(navigationView!!)
//            } else {
//                drawerLayout?.openDrawer(navigationView!!)
//            }
//        }
//
//        navigationView?.setNavigationItemSelectedListener { item: MenuItem ->
//            when (item.itemId) {
//                // Handle menu clicks here (example)
//                R.id.main -> {
//                    // Perform action
//                    true
//                }
//                else -> false
//            }
//        }
//    }
//
//    override fun onDestroy() {
//        super.onDestroy()
//        binding = null  // Prevent memory leaks
//    }
//}























